src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -116,21 +115,20 @@ public KafkaTopicRepository(final KafkaZookeeper kafkaZookeeper,
             final BatchItem item,
             final HystrixKafkaCircuitBreaker circuitBreaker,
             final Span currentSpan) throws EventPublishingException {
-        final Scope publishScope = TracingService.activateSpan(currentSpan, false);
+        final Span publishSpan = TracingService.activateSpan(currentSpan, false);
         try {
             final CompletableFuture<Exception> result = new CompletableFuture<>();
             final ProducerRecord<String, String> kafkaRecord = new ProducerRecord<>(
                     topicId,
                     KafkaCursor.toKafkaPartition(item.getPartition()),
                     item.getEventKey(),
                     item.dumpEventToString());
-
             circuitBreaker.markStart();
             producer.send(kafkaRecord, ((metadata, exception) -> {
-                final Scope producerScope = TracingService.activateSpan(currentSpan, false);
+                final Span producerSpan = TracingService.activateSpan(currentSpan, false);
                 if (null != exception) {
                     LOG.warn("Failed to publish to kafka topic {}", topicId, exception);
-                    handlePublishingException(producerScope, item, "internal error");
+                    handlePublishingException(producerSpan, item, "internal error");
```

Comment:
in my opinion this is too much, the user looking in lightstep will have no idea what happened, we need to clean it up here